docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,366 @@
+Notes on OE SDK Attestation API and Plugin Library Design
+====
+
+This document contains notes on the concepts and assumptions behind the design
+of the TEE-agnostic OE SDK Attestation API as described in document
+[Proposal of OE SDK Attestation Public and Plugin API for V0.10 Release](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/Attestation_API_Proposal.md).
+It also captures notes on the design of SGX plugins as explained in document
+[Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/SGX_QuoteEx_Integration.md).
+
+Hopefully these notes can help readers' understanding of the API and its
+usage, and the SGX plugin design.
+
+# Notes on TEE-agnostic OE SDK Attestation API Design
+
+## Semantics of Custom Claims and Optional Parameters
+
+This section explains the custom claims vs optional parameters for the [OE SDK attestation API](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/Attestation_API_Proposal.md#proposed-oe-sdk-v010-public-api)
+function `oe_get_evidence()`.
+
+Among all the parameters for the OE SDK attestation API
+function `oe_get_evidence()`,
+two of them are a list of custom claims (`custom_claims`) and an optional
+parameters blob(`opt_params`). While in general the fields in these two
+parameters fit the definition of the term "Claims" defined in the
+[RATS document](https://tools.ietf.org/html/draft-ietf-rats-architecture),
+in the context of the function `oe_get_evidence()`, these two parameters
+have distinct semantics that might not be straightforward to application
+and plugin software developers.
+
+To help clarify what goes into the custom claims (`custom_claims`) and
+what goes into the optional parameters (`opt_params`) in the function
+`oe_get_evidence()`, it would be good to articulate the semantics of these
+two entities.
+
+- Custom claims have well-defined semantics:
+  - Custom claims can be any flat array of bytes supplied by attester
+  applications.
+  - Custom claims are only meaningful to attester / verifier applications.
+  They are not inspected or interpreted by the OE SDK or its plugins.
+  - They are always TEE agnostic
+  - Custom claims are included and signed in the evidence as clear text
```

Comment:
```suggestion
  - Custom claims are included and signed in the evidence
```
Remove "as clear text", since some evidence formats do encryption or some other form of encoding for which they would not appear as clear text. Or would also be ok to do

```suggestion
  - Custom claims are included and signed in the evidence (as clear text, for the current SGX formats)
```

Reply:
Thanks!